packages/forklift-console-plugin/src/modules/Providers/hooks/useProvidersInventoryList.ts

```diff
@@ -32,29 +39,42 @@ interface UseInventoryResult {
 }
 
 /**
- * A React hook to fetch and maintain an up-to-date list of providers' inventory data.
+ * A React hook to fetch and maintain an up-to-date list of providers' inventory data, belongs to a given namespace or to all namespaces
+ * (based on the namespace parameter).
+ * For users with limited namespaces privileges, only the given namespace's providers inventory data are fetched.
  * It fetches data on mount and then at the specified interval.
  *
  * @param {UseInventoryParams} params - Configuration parameters for the hook.
+ * @param {string} namespace - namespace to fetch providers' inventory data for. if set to null, then fetch for all namespaces.
  * @param {number} [params.interval=10000] - Interval (in milliseconds) to fetch new data at.
  *
  * @returns {UseInventoryResult} result - Contains the inventory data, the loading state, and the error state.
  */
 export const useProvidersInventoryList = ({
+  namespace = null,
   interval = 20000,
 }: UseInventoryParams): UseInventoryResult => {
   const [inventory, setInventory] = useState<ProvidersInventoryList | null>(null);
   const [loading, setLoading] = useState<boolean>(true);
   const [error, setError] = useState<Error | null>(null);
   const oldDataRef = useRef(null);
   const oldErrorRef = useRef(null);
+  const canList: boolean = useFlag('CAN_LIST_NS');
+
+  // Sanity check: if user does not have permissions to list all namespaces, exit now
+  if (!canList && namespace == null) {
+    const error = new Error("user doesn't have permissions to list all namespaces");
+    handleError(error);
+
+    return { inventory, loading: false, error };
+  }
```

Comment:
we need to remove this test, because we do it before a hook ( useEffect )

Reply:
I tried to only remove the `return` statement in line 69 but leave the error handling as is, since it won't change the main flow but will just set another more clear error message. But it was also rejected by the Code Analysis....